cardano-node/src/Cardano/Tracing/Tracers.hs

```diff
@@ -634,6 +637,36 @@ mkConsensusTracers mbEKGDirect trSel verb tr nodeKern fStats = do
      when (isRollForward ev) $
        sendEKGDirectCounter ekgDirect "cardano.node.metrics.served.header.counter.int"
 
+traceBlockFetchClientMetrics
+  :: forall blk remotePeer.
+     ( )
+  => Maybe EKGDirect
+  -> STM.TVar SlotNo
+  -> Tracer IO (TraceLabelPeer remotePeer (TraceFetchClientState (Header blk)))
+  -> Tracer IO (TraceLabelPeer remotePeer (TraceFetchClientState (Header blk)))
+traceBlockFetchClientMetrics Nothing _ tracer = tracer
+traceBlockFetchClientMetrics (Just ekgDirect) lastSlotNoVar tracer = Tracer $ bfTracer
+
+  where
+    bfTracer :: TraceLabelPeer remotePeer (TraceFetchClientState (Header blk)) -> IO ()
+    bfTracer e@(TraceLabelPeer _ (CompletedBlockFetch p _ _ _ delay)) = do
+      traceWith tracer e
+      let slotNo = case pointSlot p of
+                        Origin -> SlotNo 0 -- We don't care
+                        At s   -> s
+      fresh <- atomically $ do
+          lastSlotNo <- STM.readTVar lastSlotNoVar
+          if lastSlotNo < slotNo
+             then do
+                 STM.writeTVar lastSlotNoVar slotNo
+                 return True
+             else return False
+      when fresh $
+        sendEKGDirectDouble ekgDirect "cardano.node.metrics.blockfetchclient.blockdelay.s"
```

Comment:
Why is this an EKG label metric rather than a guage metric (e.g. measured in microseconds or something)?

And do we really have to do an EKG store string lookup every time? :grimacing: Maybe that's just the way it is until we replace the EKG metrics reporting scheme.

Reply:
Doubles can't be represented directly, so we have to dumb them down to strings -- gauges would be nicer, as they are integral.